### DISCUSSION TITLE
Struktura frontendu

### DISCUSSION THREAD

**Discussion by itchystone** [emkarcinos/WMIAdventure#417] [Codding]
Chciałem zrobić komponent `Button`, który będzie używany w całej aplikacji, żeby nie powtarzać się w kółko, ale nie wiedziałem gdzie to zrobić. Zaczałem się zastanawiać, czy struktura frontendu nie powinna wyglądać inaczej, bo teraz w `atoms` mamy takie komponenty:

![image](https://user-images.githubusercontent.com/43492764/126865262-dfa0c320-ac22-47ea-8ff2-9ed3307eae4b.png)

Dla każdego atomu, jak potrzebny jest np. button to tworzymy go w tym atomie:

![image](https://user-images.githubusercontent.com/43492764/126865281-176c4724-f289-4c1b-8399-9b79960ec6e0.png)

I w ten sposób powtarzamy kod. W niektórych przyadkach to prawda, że potrzebny jest unikalny `button`, ale w takim razie gdzie umieścić  taki typowy `button`, który jest taki sam dla całości aplikacji.

Wedle tego [źródła](https://medium.com/@janelle.wg/atomic-design-pattern-how-to-structure-your-react-application-2bb4d9ca5f97) atomy to powinny być właśnie takie rzeczy jak buttony, labele itp

![image](https://user-images.githubusercontent.com/43492764/126865498-0e5e3d7a-1353-4eee-9704-96f8d9451fb5.png)

**Comment by mattyl006** [comment]
Katalog global można zrobić i tam takie komponenty powtarzalne robić. Już jest taki katalog ale w złym miejscu. Można go przenieść

**Reply by itchystone** [reply]
O spoko, a jak się nazywa?

**Reply by mattyl006** [reply]
**global** i jest chyba w *entertaiment/*, możemy go przenieść do pages/ i tworzyć tam globalne atomy, molekuły i organizmy dla każdej strefy

**Comment by mattyl006** [comment]


**Reply by itchystone** [reply]
To chyba nie ta dyskusja ^^
